algorithms/kernel/kernel_function/oneapi/kernel_function_linear_dense_default_oneapi_impl.i

```diff
@@ -0,0 +1,124 @@
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+//++
+//  Linear kernel functions implementation
+//--
+*/
+
+#ifndef __KERNEL_FUNCTION_LINEAR_DENSE_DEFAULT_ONEAPI_IMPL_I__
+#define __KERNEL_FUNCTION_LINEAR_DENSE_DEFAULT_ONEAPI_IMPL_I__
+
+#include "algorithms/kernel_function/kernel_function_types_linear.h"
+
+#include "externals/service_stat.h"
+#include "algorithms/kernel/service_error_handling.h"
+#include "algorithms/kernel/kernel_function/oneapi/cl_kernels/kernel_function.cl"
+#include "externals/service_ittnotify.h"
+#include "service/kernel/oneapi/blas_gpu.h"
+#include "service/kernel/oneapi/reducer.h"
+
+namespace daal
+{
+namespace algorithms
+{
+namespace kernel_function
+{
+namespace linear
+{
+namespace internal
+{
+using namespace daal::oneapi::internal;
+
+template <typename algorithmFPType>
+services::Status KernelImplLinearOneAPI<defaultDense, algorithmFPType>::computeInternalVectorVector(NumericTable & a1, NumericTable & a2,
+                                                                                                    NumericTable & r, const ParameterBase * par)
+{
+    return services::ErrorMethodNotImplemented;
+}
+
+template <typename algorithmFPType>
+services::Status KernelImplLinearOneAPI<defaultDense, algorithmFPType>::computeInternalMatrixVector(NumericTable & a1, NumericTable & a2,
+                                                                                                    NumericTable & r, const ParameterBase * par)
+{
+    return services::ErrorMethodNotImplemented;
+}
+
+template <typename algorithmFPType>
+services::Status KernelImplLinearOneAPI<defaultDense, algorithmFPType>::computeInternalMatrixMatrix(NumericTable & a1, NumericTable & a2,
+                                                                                                    NumericTable & r, const ParameterBase * par)
+{
+    services::Status status;
+
+    auto & context    = services::Environment::getInstance()->getDefaultExecutionContext();
+    auto & deviceInfo = context.getInfoDevice();
+
+    const size_t nVectors1 = a1.getNumberOfRows();
+    const size_t nVectors2 = a2.getNumberOfRows();
+
+    const size_t nFeatures1 = a1.getNumberOfColumns();
+    const size_t nFeatures2 = a2.getNumberOfColumns();
+    DAAL_ASSERT(nFeatures1 == nFeatures2);
+
+    const Parameter * linPar    = static_cast<const Parameter *>(par);
+    const algorithmFPType alpha = algorithmFPType(linPar->k);
```

Comment:
Please check that `linPar` is not `nullptr`

Reply:
I agree with you, but this will lead to very big changes for CPU consistency. So, I propose to leave it this way.